docs/source/tutorials_source/package/tutorial_pretrain_detectron2.py

```diff
@@ -0,0 +1,265 @@
+).to(device)
+
+# %%
+# Finally, let's build SimCLR around the backbone as shown in the other
+# tutorials. For this, we only require an additional projection head.
+projection_head = lightly.models.modules.SimCLRProjectionHead(
+    input_dim=num_ftrs,
+    hidden_dim=num_ftrs,
+    output_dim=128,
+).to(device)
+
+# %%
+# Setup data augmentations and loaders
+# ------------------------------------
+#
+# We start by defining the augmentations which should be used for training.
+# We use the same ones as in the SimCLR paper but change the input size and
+# minimum scale of the random crop to adjust to our dataset. 
+#
+# We don't go into detail here about using the optimal augmentations.
+# You can learn more about the different augmentations and learned invariances
+# here: :ref:`lightly-advanced`.
+collate_fn = lightly.data.SimCLRCollateFunction(input_size=input_size)
+
+dataset_train_simclr = lightly.data.LightlyDataset(input_dir=data_path)
+
+dataloader_train_simclr = torch.utils.data.DataLoader(
+    dataset_train_simclr,
+    batch_size=batch_size,
+    shuffle=True,
+    collate_fn=collate_fn,
+    drop_last=True,
+    num_workers=num_workers
+)
+
+# %%
+# Self-supervised pre-training
+# -----------------------------
+# Now all we need to do is define a loss and optimizer and start training!
+
+criterion = lightly.loss.NTXentLoss()
+optimizer = torch.optim.SGD(
+    list(backbone.parameters()) + list(projection_head.parameters()),
+    lr=lr,
+    momentum=0.9,
+    weight_decay=1e-6,
+)
+
+
+for e in range(max_epochs):
+
+    mean_loss = 0.
+    for (x0, x1), _, _ in dataloader_train_simclr:
+
+        x0 = x0.to(device)
+        x1 = x1.to(device)
+
+        y0 = projection_head(backbone(x0).flatten(start_dim=1))
+        y1 = projection_head(backbone(x1).flatten(start_dim=1))
+
+        # backpropagation
+        loss = criterion(y0, y1)
+        loss.backward()
+
+        optimizer.step()
+        optimizer.zero_grad()
+
+        # update average loss
+        mean_loss += loss.detach().cpu().item() / len(dataloader_train_simclr)
+
+    print(f'[Epoch {e:2d}] Mean Loss = {mean_loss:.2f}')
+
+
+# %%
+# Storing the checkpoint
+# -----------------------
+# Now, we can use the pre-trained backbone from the Detectron2 model. The code
+# below shows how to save it as a Detectron2 checkpoint called `my_model.pth`.
+
+# get the first module from the backbone (i.e. the detectron2 ResNet)
+detmodel.backbone.bottom_up = backbone[0]
+
+checkpointer = DetectionCheckpointer(detmodel, save_dir='./')
+checkpointer.save('my_model')
+
+
+# %%
+# Finetuning with Detectron2
+# ---------------------------
+#
+# The checkpoint from above can now be used by any Detectron2 script. For example,
+# you can use the `train_net.py` script in the Detectron2 `tools`:
+#
+#
+
+# %%
+#.. code-block:: none
+#
+#   python train_net.py --config-file ../configs/COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_1x.yaml \
+#       MODEL.WEIGHTS path/to/my_model.pth \
```

Comment:
The `path/to/my_model.pth` should match the one you set in line 225/226 in my eyes (`./my_model.pth`).
Perhaps you could use e.g. `path/to/checkpoint` as `save_dir` in line 225.

Reply:
the `train_net.py` is in the detectron repository so it doesn'tm ake sense to use the same path twice.